File: Problem-7---Turbo-Sort/Main.c
#include <stdio.h>
#include <stdlib.h>

int arr[1000001] = {0};

int main()
{
	//printf("%s\n", "Enter the number of test cases");
	int num_of_test_cases;
	scanf("%d", &num_of_test_cases);
	int num,i;
	for (i = 0; i < num_of_test_cases; i++)
	{
		scanf("%d", &num);
		arr[num]++;
	}

	for (i = 0; i < 1000001; i++)
	{
		while(arr[i] > 0)
		{
			printf("%d\n", i);
			arr[i]--;
		}
	}
	return 0;
}

File: Problem-7---Turbo-Sort/Main.rb
def tsort(tests)
	sorted_array = Array.new(1000001){0}
	1.upto(tests) do
		input_num = gets.chomp.to_i
		sorted_array[input_num] += 1
	end
	#puts sorted_array.sort!
	sorted_array.each_index {|index| puts index if sorted_array[index] != 0}
end

#puts "enter the number of test cases"
test_cases = gets.chomp.to_i
tsort(test_cases)
